mailpoet/assets/js/src/automation/editor/components/header/errors.tsx

```diff
@@ -0,0 +1,169 @@
+  const compositeState = useContext(ErrorsCompositeContext);
+
+  const { steps, workflowData } = useSelect(
+    (select) => ({
+      steps: select(storeName).getSteps(),
+      workflowData: select(storeName).getWorkflowData(),
+    }),
+    [],
+  );
+
+  const { openSidebar, selectStep } = useDispatch(storeName);
+
+  const stepData = workflowData.steps[stepId];
+  const step = steps.find(({ key }) => key === stepData.key);
+
+  return (
+    <CompositeItem
+      className="mailpoet-automation-step-error"
+      role="listitem"
+      state={compositeState}
+      onClick={() => {
+        openSidebar(stepSidebarKey);
+        selectStep(stepData);
+      }}
+    >
+      <ColoredIcon
+        icon={step.icon}
+        foreground={step.foreground}
+        background={step.background}
+        width="23px"
+        height="23px"
+      />
+      {step.title}
+    </CompositeItem>
+  );
+}
+
+export function Errors(): JSX.Element | null {
+  const [showPopover, setShowPopover] = useState(false);
+
+  const compositeState = useCompositeState({
+    orientation: 'vertical',
+    shift: true,
+  });
+
+  const { errors, workflowData } = useSelect(
+    (select) => ({
+      errors: select(storeName).getErrors(),
+      workflowData: select(storeName).getWorkflowData(),
+    }),
+    [],
+  );
+
+  // walk the steps tree (breadth first) to produce stable error order
+  const stepErrors = useMemo(() => {
+    if (!errors) {
+      return [];
+    }
+
+    const visited = new Map<string, StepErrorType | undefined>();
+    const ids = workflowData.steps.root.next_steps.map(({ id }) => id);
+    while (ids.length > 0) {
+      const id = ids.shift();
+      if (!visited.has(id)) {
+        visited.set(id, errors.steps[id]);
+        workflowData.steps[id]?.next_steps?.forEach((step) =>
+          ids.push(step.id),
+        );
+      }
+    }
+    return [...visited.values()].filter((error) => !!error);
+  }, [errors, workflowData]);
+
+  // automatically open the popover when errors appear
+  const hasErrors = stepErrors.length > 0;
+  useEffect(() => {
+    if (hasErrors) {
+      setShowPopover(true);
+    }
+  }, [hasErrors]);
+
+  if (stepErrors.length === 0) {
+    return null;
+  }
+
+  return (
+    <div>
+      <Button
+        variant="link"
+        onClick={() =>
+          setShowPopover((prevState) =>
+            prevState === undefined ? false : !prevState,
+          )
+        }
+        onMouseDown={() =>
+          // Catch and mark a mouse down event from an open popover with "undefined" to avoid closing it
+          // (automatically via click outside) and reopening it right after (via the onClick handler).
+          // The "onClose" method of the popover doesn't pass any events so we can't filter them.
+          setShowPopover((prevState) => (prevState ? undefined : prevState))
+        }
+        style={{ textDecoration: 'none', borderRadius: 99999 }}
+      >
+        <Chip>{stepErrors.length} issues</Chip>
```

Comment:
`_n( '1 issue', '%d issues', stepErrors.length, 'mailpoet' );`

So, we have an instance where WP translations would be really useful :)